### PR TITLE
Fix(CheckBoxFields) add check for any empty checkbox fields when saving a record

### PR DESCRIPTION
**Pull request by stephenmcm**
Now includes tests. I've removed the older Travis env's as they are fail for the already included tests.

### CODE REVIEW

**Comment by silbinarywolf** [comment]
@stephenmcm Can you please update the README.md to say "PHP 5.6+" under supports?
Since updating .travis.yml to not test PHP 5.5 or below means that we won't *technically* support it.

Once that's done, I'll merge it in.

**Comment by stephenmcm** [comment]
All done, now.

File: tests/MultiRecordSaveAndEditTest.php [hunk 1]
```diff
@@ -28,6 +28,14 @@ public function testSaveAndEditNewRecords_NoSort()
             "admin/pages/edit/EditForm/field/"."HasManyRelation"."/addinlinerecord/MultiRecordField_HasManyTest/new/field/".key($_FILES)."/upload",
             $_FILES
         );
+        if ($response->getBody() == "I can't handle sub-URLs on class CMSPageEditController.") {
+            // Note Stephen (2018-05-29) The current SilverStripe 3 branch expects the following url format instead.
+            // It includes the page ID in the URL e.g: "admin/pages/edit/EditForm/1/field/"
```
Comment (silbinarywolf) [inline tests/MultiRecordSaveAndEditTest.php]: This comment doesn't accurately describe the problem.
Can you adjust slightly to say something like:
"In versions SilverStripe 3.6 and lower, there is no page ID passed to an EditForm action. However in 3.7 (`3` branch at time of writing), this has been changed"

**Comment by silbinarywolf** [comment]
Dropping PHP 5.3 is fine because Travis doesnt' support it anymore AFAIK!
Please just make a minor change to the comment above and I'll merge it in!

Thanks for your work and adding the test :D

**Pull request merged**